File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []

        def helper(root, level):
            if root:
                if level < len(res):
                    res[level].append(root.val)
                else:
                    res.append([root.val])
                helper(root.left, level + 1)
                helper(root.right, level + 1)
            
        helper(root, 0)
        res = [level[-1] for level in res]
        return res

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        return maxDepthH(root, 0);
    }

    public int maxDepthH(TreeNode current, int depth) {
        if (current == null) {
            return depth;
        }
        else {
            return Math.max(maxDepthH(current.left, depth + 1), maxDepthH(current.right, depth + 1));
        }
    }
}
